handbook/engineering/search/index.md

```diff
@@ -2,9 +2,93 @@
 
 We own the end-to-end Sourcegraph search experience.
 
+## Vision
+
+Sourcegraph search is:
+
+- **Fast:** Performance of showing results is fast and I can get to what I’m looking for quickly.
+- **Expressive:** It is possible to construct a search query to find exactly the results I’m looking for. The tools, syntax, filters, are provided to construct the queries I want to write.
+The answer to “Can I do ‘X’ with Sourcegraph?” will always be yes.
+- **Easy to use:** Users can figure out how to find what they are looking for, the syntax is clear, new users understand what options they have for searching.
```

Comment:
> Users can figure out how to find what they are looking for,

This feels a bit vague, and already true. If users dug deep enough into our docs, they would indeed be able to find out how to use our search. I think the goal is to make our product intuitive enough that new users know what to do on their first visit.

The rest of it, I really like. Perhaps a rephrasing:

```suggestion
- **Easy to use:**  Users quickly understand how to conduct a search and what options are available for searching. The search syntax is clear and intuitive.
```

Reply:
I really like this suggestion:
_The search syntax is clear and intuitive._ 

Some parts of this were left open-ended intentionally: 

> How to find

through conducting a search, using the search results UI more effectively, consuming multiple notification channels

> What they are looking for

code, repositories, files, metadata, etc.

What do you think of this rephrasing:
Users can quickly understand how to find what they are looking for and what options are available for searching. The search syntax is clear and intuitive.